src/rgw/rgw_kms.cc

```diff
@@ -23,6 +24,15 @@ static map<string,string> get_str_map(const string &str) {
   return m;
 }
 
+static void concat_url(std::string &url, std::string path) {
+  if (url.back() == '/' && path.front() == '/') {
+    url.pop_back();
+  } else if (url.back() != '/' && path.front() != '/') {
+    url.push_back('/');
+  }
+  url.append(path);
+}
+
```

Comment:
Not ideal to have to write this sort of function here but I've seen this kind of code scattered around the code base in different forms and didn't find any suitable functions I could reuse.